packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -0,0 +1,829 @@
+      userId,
+      password,
+    });
+
+    try {
+      const secretData = await this.toprfClient.fetchAllSecretDataItems({
+        decKey: encKey,
+        authKeyPair,
+      });
+
+      if (secretData?.length > 0) {
+        await this.#createNewVaultWithAuthData({
+          password,
+          rawToprfEncryptionKey: encKey,
+          rawToprfAuthKeyPair: authKeyPair,
+        });
+      }
+
+      return this.#withPersistedSeedPhraseBackupsState<Uint8Array[]>(() =>
+        Promise.resolve(
+          SeedPhraseMetadata.parseSeedPhraseFromMetadataStore(secretData),
+        ),
+      );
+    } catch (error) {
+      log('Error fetching seed phrase metadata', error);
+      throw new Error(
+        SeedlessOnboardingControllerError.FailedToFetchSeedPhraseMetadata,
+      );
+    }
+  }
+
+  /**
+   * @description Update the password of the seedless onboarding flow.
+   *
+   * Changing password will also update the encryption key and metadata store with new encrypted values.
+   *
+   * @param params - The parameters for updating the password.
+   * @param params.authConnectionId - OAuth authConnectionId from dashboard
+   * @param params.groupedAuthConnectionId - Optional grouped authConnectionId to be used for the authenticate request.
+   * You can pass this to use aggregate connection.
+   * @param params.userId - user email or id from Social login
+   * @param params.newPassword - The new password to update.
+   * @param params.oldPassword - The old password to verify.
+   */
+  async changePassword(params: {
+    authConnectionId: string;
+    groupedAuthConnectionId?: string;
+    userId: string;
+    newPassword: string;
+    oldPassword: string;
+  }) {
+    // verify the old password of the encrypted vault
+    await this.#verifyPassword(params.oldPassword);
+
+    try {
+      // update the encryption key with new password and update the Metadata Store
+      const { encKey: newEncKey, authKeyPair: newAuthKeyPair } =
+        await this.#changeEncryptionKey(params);
+
+      // update and encrypt the vault with new password
+      await this.#createNewVaultWithAuthData({
+        password: params.newPassword,
+        rawToprfEncryptionKey: newEncKey,
+        rawToprfAuthKeyPair: newAuthKeyPair,
+      });
+    } catch (error) {
+      log('Error changing password', error);
+      throw new Error(SeedlessOnboardingControllerError.FailedToChangePassword);
+    }
+  }
+
+  /**
+   * @description Get the hash of the seed phrase backup for the given seed phrase, from the state.
+   *
+   * If the given seed phrase is not backed up and not found in the state, it will return `undefined`.
+   *
+   * @param seedPhrase - The seed phrase to get the hash of.
+   * @returns A promise that resolves to the hash of the seed phrase backup.
+   */
+  getSeedPhraseBackupHash(seedPhrase: Uint8Array): string | undefined {
+    return this.state.backupHashes.find((hash) => {
+      return hash === keccak256AndHexify(seedPhrase);
+    });
```

Comment:
You can compute the hash once and then compare against the stored hash:
```suggestion
    const seedPhraseHash = keccak256AndHexify(seedPhrase);
    return this.state.backupHashes.find((hash) => {
      const 
      return hash === seedPhraseHash;
    });
```